src/compiler/commandLineParser.ts

```diff
@@ -1201,6 +1201,15 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
         category: Diagnostics.Modules,
         description: Diagnostics.Conditions_to_set_in_addition_to_the_resolver_specific_defaults_when_resolving_imports,
     },
+    {
+        name: "resolveSideEffectImports",
+        type: "boolean",
+        affectsSemanticDiagnostics: true,
+        affectsBuildInfo: true,
+        category: Diagnostics.Modules,
+        description: Diagnostics.Resolve_side_effect_imports,

```

Comment:
Resolve is very misleading here. We always resolve these irrespective of this flag right. We only "check" with this flag.

Reply:
Sort of; we never actually called `resolveExternalModuleName` on them before.

What would you suggest instead? checkSideEffectImports? noUncheckedSideEffectImports?

Reply:
But resolveExternalModuleName is just part of checking. We did resolve those imports and add the found files into the program so from resolution point of view nothing changed. 

vote for `noUncheckedSideEffectImports` because we have more of `noUnchecked` options .. but `check` would be ok too i guess ..